wagtail/admin/views/generic/models.py

```diff
@@ -75,10 +75,13 @@ class IndexView(
 ):
     model = None
     index_url_name = None
+    index_results_url_name = None
     add_url_name = None
     add_item_label = _("Add")
     edit_url_name = None
     template_name = "wagtailadmin/generic/index.html"
+    results_template_name = "wagtailadmin/generic/index_results.html"
+    results_only = False  # If true, just render the results as an HTML fragment
```

Comment:
Was wondering if we could replace `results_only` by overriding the `template_name` when the view is instantiated in the viewset, but then I realised the view can be used without a viewset (as is the case with the users views). I don't think it's worth having a subclass, so I think this is fine 👍

Reply:
I tried subclassing `IndexView` as `IndexResultsView` and overriding `template_name` as my first attempt at this, but it got messy - IIRC, the problem was that any existing viewset that provided its own custom IndexView subclass (e.g. to define columns) would then have to be updated with a corresponding IndexResultsView subclass with the same customisations.